Extension/src/LanguageServer/client.ts

```diff
@@ -864,6 +873,90 @@ export class DefaultClient implements Client {
         return workspaceFolder ? workspaceFolder.name : "untitled";
     }
 
+    public updateClientConfigurations(): void {
+        clients.forEach(client => {
+            if (client instanceof DefaultClient) {
+                const defaultClient: DefaultClient = <DefaultClient>client;
+                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                defaultClient.configuration.handleConfigurationChange();
+            }
+        });
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const settings: OtherSettings = new OtherSettings();
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
```

Comment:
In the "no compiler found" scenario, it's possible this code could be executed (if other workarounds are done), in which case this is undefined (the value returned from cpptools). It seems like function should be made to handle that so it can still show the 3 other options.